xla/service/cpu/onednn_ops_rewriter.cc

```diff
@@ -58,11 +60,61 @@ HloInstruction* FindLayerNormShift(HloInstruction* instr) {
   return shift;
 }
 
-std::optional<HloInstruction*> MatchSoftmax(HloInstruction* instr) {
+bool is_neg_inf_const_scalar(const HloInstruction* const_instr) {
+  if (const_instr->opcode() != HloOpcode::kConstant) {
+    return false;
+  };
+  if (!ShapeUtil::IsEffectiveScalar(const_instr->shape())) {
+    return false;
+  };
+  auto value = LiteralUtil::GetFirstScalarLiteral(const_instr->literal());
+  return literal_comparison::Equal(
+             value, LiteralUtil::MinValue(const_instr->shape().element_type()))
+      .ok();
+}
+
+// This function is to make sure the reducer computation is strictly a max
+// reducer and root instruction (kMaximum) has same inputs as computation
+bool IsMaxReducerComputation(const HloComputation* comp) {
+  if (comp->root_instruction()->opcode() != HloOpcode::kMaximum) {
+    return false;
+  }
+  auto max_instr = comp->root_instruction();
+  for (int idx = 0; idx < comp->num_parameters(); ++idx) {
+    if (comp->parameter_instruction(idx) != max_instr->operand(0) &&
+        comp->parameter_instruction(idx) != max_instr->operand(1)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// Pattern to match AnyOf  Maximum(Reduce_max(...), -inf) OR Reduce_max(...)
```

Comment:
Please end sentence with dot (`.`). (And, unless there's a justifiable reason for it, remove double space after `AnyOf`.)